core/src/main/java/org/elasticsearch/common/geo/GeoPoint.java

```diff
@@ -185,4 +186,12 @@ public static GeoPoint fromGeohash(String geohash) {
     public static GeoPoint fromGeohash(long geohashLong) {
         return new GeoPoint().resetFromGeoHash(geohashLong);
     }
+
+    public double distanceFrom(GeoPoint other) {
+        return GeoUtils.arcDistance(lat, lon, other.lat, other.lon);
+    }
+
+    public double distanceFrom(GeoPoint other, DistanceUnit units) {
```

Comment:
It doesn't look like `DistanceUnit` is available in painless (needs to be whitelisted). However, I'm also not sure why we really need this. Can we have just `distanceFrom(point)` above and leave conversion to units other than meters to the user?